File: codewars/6kyu/triangle-type/dduck-triangle-type.py
# triangle-type
# https://www.codewars.com/kata/53907ac3cd51b69f790006c5

from unittest import TestCase


def triangle_type(a, b, c):
    max_side = max(a, b, c)
    min_side = min(a, b, c)
    middle_side = sum([a, b, c]) - max_side - min_side

    sum_of_min_middle_side_square = min_side ** 2 + middle_side ** 2
    max_side_square = max_side ** 2

    if min_side + middle_side <= max_side:
        return 0
    if sum_of_min_middle_side_square == max_side_square:
        return 2
    if sum_of_min_middle_side_square > max_side_square:
        return 1
    if sum_of_min_middle_side_square < max_side_square:
        return 3

    return 0

File: codewars/6kyu/triangle-type/dduck-triangle-type.js
// triangle-type
// https://www.codewars.com/kata/53907ac3cd51b69f790006c5

/* Should return ᐃ type:
  0 : if ᐃ cannot be made with given sides
  1 : acute ᐃ
  2 : right ᐃ
  3 : obtuse ᐃ
*/
const triangleType = (a, b, c) => {
  const max = Math.max(a, b, c);
  const min = Math.min(a, b, c);
  const middle = (a + b + c) - max - min;
  
  if (min + middle <= max) {
    return 0;
  }
  
  const sumOfBCSquare = min ** 2 + middle ** 2;
  const maxSquare = max ** 2;
  
  if (sumOfBCSquare === maxSquare) {
    return 2;
  }
  if (sumOfBCSquare > maxSquare) {
    return 1;
  }
  if (sumOfBCSquare < maxSquare) {
    return 3;
  }
};
